src/NuGet.Server.V2/Controllers/NuGetODataController.cs

```diff
@@ -165,13 +165,14 @@ public virtual IHttpActionResult GetPropertyFromPackages(string propertyName, st
 
             var idValues = packageIds.Trim().Split(new[] { '|' }, StringSplitOptions.RemoveEmptyEntries);
             var versionValues = versions.Trim().Split(new[] { '|' }, StringSplitOptions.RemoveEmptyEntries);
-            var targetFrameworkValues = String.IsNullOrEmpty(targetFrameworks) ? null :
-                                                                                 targetFrameworks.Split('|').Select(VersionUtility.ParseFrameworkName).ToList();
-            var versionConstraintValues = String.IsNullOrEmpty(versionConstraints)
+            var targetFrameworkValues = String.IsNullOrEmpty(targetFrameworks) 
+                                        ? null 
+                                        : targetFrameworks.Split('|').Select(VersionUtility.ParseFrameworkName).ToList();
+            var versionConstraintValues = (String.IsNullOrEmpty(versionConstraints)
                                             ? new string[idValues.Length]
-                                            : versionConstraints.Split('|');
+                                            : versionConstraints.Split('|')).ToList();
```

Comment:
I added the ToList() because I modified code further below to remove element from the list if versionparsing failed. It was simpler to remove from a list than (versionConstraints and versions must have the same number of elements).